src/graphics/program-lib/programs/standard.js

```diff
@@ -235,12 +235,13 @@ var standard = {
     },
 
     _setMapTransform: function (codes, name, id, uv) {
-        codes[0] += "uniform vec4 texture_" + name + "MapTransform;\n";
+        const varName = `texture_${name}MapTransform`;
+        const checkId = id + uv * 100;
 
-        var checkId = id + uv * 100;
+        codes[0] += `uniform mat3 ${varName};\n`;
```

Comment:
I'd either try to store this as Vec3[2], or even simpler as two Vec3s, without an array.

Reply:
I'm not convinced optimising strictly for number of float uniforms is an overall win (at the cost of additional CPU work, for example), but I've done it that way for now.